[[ LCIDLC ]] Fix issues building lcidl based externals

montegoulding opened the pull request:
This patch is required for building a lcidlc based external with our current Werror config.

montegoulding committed:
[[ Cleanup ]] Add missing field initializer

montegoulding committed:
[[ LCIDLC ]] Fixes for building on windows

montegoulding committed:
[[ Lcidlc ]] Allocate space for null terminator

This patch adds an extra char to the `malloc` call in
`LCExceptionRaise` because `snprintf` returns the length
excluding the null terminator.

livecodeali commented:
@livecode-vulcan review ok cac4fed

livecode-vulcan commented:
💙 review by @livecodeali ok cac4fed6c457599803eaed50add3d47796f8f419

livecode-vulcan commented:
😞 test failure cac4fed6c457599803eaed50add3d47796f8f419

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/934)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/950)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/924)
* try-community-universal-ios-iphoneos11.0: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.0/builds/99)
* try-community-universal-ios-iphonesimulator11.0: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.0/builds/100)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/928)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/950)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/934)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/950)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/933)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/943)
* try-community-x86-win32: [failure](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/965)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/929)

livecodeali commented:
@livecode-vulcan retry cac4fed

montegoulding commented:
Hmm... Retry seems to be ignored.

montegoulding commented:
@livecode-vulcan retry cac4fed

montegoulding committed:
[[ Bug ]] Check correct var

When optional c-strings are not sent to a lcidlc external the
`default__cstring` function is called with the default value. Because it
was checking the wrong var it was always throwing an out of memory
error.

livecodeali commented:
@livecode-vulcan review ok cf6b6a4

livecode-vulcan commented:
💙 review by @livecodeali ok cf6b6a4680299ebe05cd59fcf71460de53a2c8af

livecode-vulcan commented:
😎 test success cf6b6a4680299ebe05cd59fcf71460de53a2c8af

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/940)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/956)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/930)
* try-community-universal-ios-iphoneos11.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.1/builds/3)
* try-community-universal-ios-iphonesimulator11.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.1/builds/3)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/934)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/957)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/940)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/956)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/939)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/949)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/971)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/935)
